app/views/shared/_navigation_bar.html.erb

```diff
@@ -1,21 +1,38 @@
 <nav class="navbar navbar-expand-lg navbar-dark bg-primary justify-content-between" data-turbolinks-animate-persist="true">
   <a class="navbar-brand" href="#"><i class="fas fa-sitemap"></i> Caucus</a>
-  <div class="collapse navbar-collapse" id="navbarToggler" style="justify-content: right;">
-    <% if user_signed_in? && current_user.has_role?(:admin) %>
-    <div class="navbar-nav mr-4">
-      <%= link_to "Admin", admin_categories_path, class: "nav-link" %>
-    </div>
-    <% end %>
-    <div class="navbar-nav">
-      <% if user_signed_in? %>
-        <%= link_to("Instructions", instructions_path, class: "nav-link") %>
-        <%= link_to("Logout", destroy_user_session_path, method: :delete, class: "nav-link") %>
-      <% else %>
-        <%= link_to("Login", new_user_session_path, class: "nav-link") %>
-      <% end %>
-    </div>
-  </div>
   <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarToggler" aria-controls="navbarToggler" aria-expanded="false" aria-label="Toggle navigation">
     <span class="navbar-toggler-icon"></span>
   </button>
+
+  <div class="collapse navbar-collapse" id="navbarToggler" style="justify-content: right;">
+  <% if user_signed_in? %>
+  <div class="navbar-nav mr-1">
+    <%= image_tag current_user.avatar.variant(resize_to_limit: [25, 25]) if current_user.avatar.attached? %>
```

Comment:
Ignorant question: this is just picking the most appropriate avatar and displaying it, right? It's not live-generating a specific image, is it?

Reply:
It's all under the hood railsy-magic. How it handles it specifically I'm unsure, but I pulled this example straight from the docs.